File: app/src/main/java/com/asfoundation/wallet/interact/FetchGasSettingsInteract.java
package com.asfoundation.wallet.interact;

import com.asfoundation.wallet.entity.GasSettings;
import com.asfoundation.wallet.repository.GasSettingsRepositoryType;
import io.reactivex.Single;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class FetchGasSettingsInteract {
  private final GasSettingsRepositoryType repository;

  public FetchGasSettingsInteract(GasSettingsRepositoryType repository) {
    this.repository = repository;
  }

  public Single<GasSettings> fetch(boolean forTokenTransfer) {
    return repository.getGasSettings(forTokenTransfer)
        .subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread());
  }
}

File: app/src/main/java/com/asfoundation/wallet/interact/FetchGasSettingsInteract.kt
package com.asfoundation.wallet.interact

import com.appcoins.wallet.core.utils.android_common.RxSchedulers
import com.asfoundation.wallet.entity.GasSettings
import com.asfoundation.wallet.repository.GasSettingsRepositoryType
import io.reactivex.Single
import javax.inject.Inject

class FetchGasSettingsInteract @Inject constructor(private val repository: GasSettingsRepositoryType,
                               private val rxSchedulers: RxSchedulers
) {

  fun fetch(forTokenTransfer: Boolean): Single<GasSettings> {
    return repository.getGasSettings(forTokenTransfer)
        .subscribeOn(rxSchedulers.io)
        .observeOn(rxSchedulers.main)
  }
}